expression/builtin_string.go

```diff
@@ -3951,6 +3951,16 @@ func (b *builtinWeightStringSig) evalString(row chunk.Row) (string, bool, error)
 			b.ctx.GetSessionVars().StmtCtx.AppendWarning(errTruncatedWrongValue.GenWithStackByArgs(tpInfo, str))
 			str = str[:b.length]
 		} else if b.length > lenStr {
+			sc := b.ctx.GetSessionVars().StmtCtx
+			valStr, _ := b.ctx.GetSessionVars().GetSystemVar(variable.MaxAllowedPacket)
```

Comment:
We should save maxAllowedPacket in Sig and set it in getFunction. (see  builtinRepeatSig)

Reply:
OK, I have done this.